munimap/query/features.py

```diff
@@ -47,8 +47,13 @@ def key(feature):
                 v = v.replace('ö', 'oe')
                 v = v.replace('ü', 'ue')
                 v = v.replace('ß', 'ss')
-                v = tuple(locale.strxfrm(str(p)) for p in
-                    v.encode(locale_encoding, errors='replace')).split(' ')
+                v = tuple(locale.strxfrm(str(encoded_v)) for encoded_v in
+                    v.encode(locale_encoding, errors='replace').split(b' '))
```

Comment:
Why is `v` encoded and then decoded again? Calling `str()` should just decode the bytes again?
I suppose the `errors='replace' will filter out encoding errors. If that is the reason, then I would do something like:
```
v = str(v.encode(locale_encoding, errors='replace'))
```
this makes it more clear what is happening and why it is done.

Reply:
then of course the split would need to be changed back to `.split(' ')`

Reply:
maybe even `v.encode(...).decode(...)`

Reply:
from the docs:

> if at least one of encoding or errors is given, object should be a [bytes-like object](https://docs.python.org/3/glossary.html#term-bytes-like-object) (e.g. [bytes](https://docs.python.org/3/library/stdtypes.html#bytes) or [bytearray](https://docs.python.org/3/library/stdtypes.html#bytearray)). In this case, if object is a [bytes](https://docs.python.org/3/library/stdtypes.html#bytes) (or [bytearray](https://docs.python.org/3/library/stdtypes.html#bytearray)) object, then str(bytes, encoding, errors) is equivalent to [bytes.decode(encoding, errors)](https://docs.python.org/3/library/stdtypes.html#bytes.decode). Otherwise, the bytes object underlying the buffer object is obtained before calling [bytes.decode()](https://docs.python.org/3/library/stdtypes.html#bytes.decode).

Reply:
> Passing a [bytes](https://docs.python.org/3/library/stdtypes.html#bytes) object to [str()](https://docs.python.org/3/library/stdtypes.html#str) without the encoding or errors arguments falls under the first case of returning the informal string representation (see also the [-b](https://docs.python.org/3/using/cmdline.html#cmdoption-b) command-line option to Python)

https://docs.python.org/3/library/stdtypes.html#str